scripts/dts/gen_defines.py

```diff
@@ -570,34 +570,38 @@ def write_phandle_val_list_entry(node, entry, i, ident):
     initializer_vals = []
     if entry.controller.label is not None:
         ctrl_ident = ident + "_CONTROLLER"  # e.g. PWMS_CONTROLLER
+        alias = None
         if entry.name:
-            ctrl_ident = str2ident(entry.name) + "_" + ctrl_ident
+            alias = str2ident(entry.name) + "_" + ctrl_ident
```

Comment:
Could put ``alias`` closer to where it's used, rename it to ``name_alias``, and pass it directly instead of as a keyword argument.

```python

if entry.name:
    name_alias = str2ident(entry.name) + "_" + ctrl_ident
else:
    name_alias = None

out_dev_s(node, ctrl_ident, entry.controller.label, name_alias)
```

(and similarly in the other spots).

Reply:
Oh... wait, this'll get buggy, because the `name_alias` variable is reused later.

Reply:
Or nah, it won't, because it gets set in both paths.

Reply:
Updated